### DISCUSSION TITLE
Why packages are store inside local node_modules folder?

### DISCUSSION THREAD

**Discussion by DanielRios549** [pnpm/pnpm#3533] [Q&A] [answered]
I have an old project that still uses NPM, and I am testing different package managers to conclude what is the best option to upgrade from the current NPM.

I like too much the purpose of PNPM, use symlinks to the same packages and versions, but when I install the packages with `pnpm i` using the `package.json` file, the packages went all stored inside the local `node_modules` folder on a subfolder called `.pnpm`, and the symlinks are all pointing to this folder.

This simply makes no sense for me, the packages should be stored on `~/.pnpm-store` folder, not inside project one. Why did this happen to me? Is it the default behavior?

![image](https://user-images.githubusercontent.com/11152535/121793961-7c151e80-cbda-11eb-8c8c-c18c13c10927.png)

**Comment by zkochan** [comment, marked as answer]
See https://pnpm.io/faq#why-does-my-node_modules-folder-use-disk-space-if-packages-are-stored-in-a-global-store

> pnpm creates hard links from the global store to the project's node_modules folders. Hard links point to the same place on the disk where the original files are. So, for example, if you have foo in your project as a dependency and it occupies 1MB of space, then it will look like it occupies 1MB of space in the project's node_modules folder and the same amount of space in the global store. However, that 1MB is the same space on the disk addressed from two different locations. So in total foo occupies 1MB, not 2MB.

**Reply by DanielRios549** [reply]
I think I'm understanding, Nautilus does not show if the file is a hard link, only if it is a symbolic link. And hard links are not allowed with folders, so the folders really exist inside `node_modeles/.pnpm`, but the files are hard links to the global store, am I right?

**Reply by zkochan** [reply]
Correct, folders are real. Also commands shims in `.bin` folders are real. Everything else is either a hard link to the global content-addressable store or a symlink to a package in `node_modules/.pnpm`
